helix-term/src/ui/prompt.rs

```diff
@@ -191,6 +194,13 @@ impl Prompt {
     }
 
     pub fn insert_char(&mut self, c: char, cx: &Context) {
+        if let Some(handler) = &self.next_char_handler.take() {
+            handler(self, c, cx);
+
+            self.next_char_handler = None;
+            return;
+        }
```

Comment:
This affects the file picker prompt as well without any register display showing up.

Reply:
yes, I think we want the feature there too. Regarding the info showing, I haven't figured out a way to show it as the completion is clipped and the authinfo is overlapped for pickers.

Reply:
I think currently it should either `ctrl-r` be disabled for file picker or make it work for file picker, otherwise it might be surprising for others to not see any feedback when a key is press after `ctrl-r`.

Reply:
The function works except the completions are not showing. I would suggest we keep it as it's useful before we are able to figure out a way to  show help info for pickers.